packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,917 @@
+            ] as const;
+
+            await element.setData(data);
+            await connect();
+            await waitForUpdatesAsync();
+        }
+
+        async function setupTableWithGrouping(): Promise<void> {
+            const data: readonly SimpleTableRecord[] = [
+                {
+                    id: '1',
+                    stringData1: 'a1',
+                    stringData2: 'b1',
+                    stringData3: 'c1'
+                },
+                {
+                    id: '2',
+                    stringData1: 'a2',
+                    stringData2: 'b1',
+                    stringData3: 'c2'
+                },
+                {
+                    id: '3',
+                    stringData1: 'a3',
+                    stringData2: 'b3',
+                    stringData3: 'c3'
+                },
+                {
+                    id: '4',
+                    stringData1: 'a4',
+                    stringData2: 'b3',
+                    stringData3: 'c4'
+                }
+            ] as const;
+
+            await element.setData(data);
+            column2.groupIndex = 0;
+            await connect();
+            await waitForUpdatesAsync();
+        }
+
+        async function setupLargeDataTable(): Promise<void> {
+            const data: SimpleTableRecord[] = [];
+            for (let i = 0; i < largeDataRowCount; i++) {
+                data.push({
+                    id: i.toString(),
+                    stringData1: `a${i}`,
+                    stringData2: `b${i}`,
+                    stringData3: `c${i}`
+                });
+            }
+
+            await element.setData(data);
+            await connect();
+            await waitForUpdatesAsync();
+        }
+
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setupWithTextColumns());
+            pageObject = new TablePageObject<SimpleTableRecord>(element);
+            column1 = element.querySelector<TableColumnText>('#first-column')!;
+            column2 = element.querySelector<TableColumnText>('#second-column')!;
+            column3 = element.querySelector<TableColumnText>('#third-column')!;
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        describe('for a simple table (no hierarchy/grouping/selection) with focus', () => {
+            beforeEach(async () => {
+                await setupBasicTable();
+                element.focus();
+                await waitForUpdatesAsync();
+            });
+
+            it('on initial table focus, the 1st column header is focused if sortable', () => {
+                const firstHeader = pageObject.getHeaderElement(0);
+                expect(currentFocusedElement()).toBe(firstHeader);
+            });
+
+            it('consecutive Enter key presses will sort the focused column (ascending, descending, no sort)', async () => {
+                const firstHeader = pageObject.getHeaderElement(0);
```

Comment:
Would this reflect our intent better if instead of retrieving the actual element (directly) that we will issue events on, we instead get `document.activeElement`? This, to me, better indicates that "on initial focus pressing `Enter` sorts a column".

Reply:
Done for these instances.